lib/ignore/ignore.go

```diff
@@ -52,6 +60,27 @@ func (p Pattern) String() string {
 	return ret
 }
 
+func (p Pattern) allowsSkippingIgnoredDirs() bool {
+	if p.result.IsIgnored() {
+		return true
+	}
+	if p.pattern[0] != '/' {
+		return false
+	}
+	// Double asterisk everywhere in the path except at the end is bad
+	tocheck := strings.TrimSuffix(p.pattern[1:], "**")
+	if strings.Contains(tocheck, "**") {
+		return false
+	}
+	// Any wildcards (*, [...]) anywhere except for the last path component are bad
```

Comment:
I think `**` is also a valid wildcard in the library.

I think all you need to do is:
1. Get the string without the last component
2. Check that quoted string matches original string

Reply:
The comment shouldn't list wildcards anymore.  
`**` is different from other wildcards, in that it also can't appear in the last component, except as a suffix (e.g. `!/foo/b**r` needs to descend to catch `foo/boo/far`), while all other wildcards are ok in the last component.